test/linkify-matrix-test.ts

```diff
@@ -0,0 +1,227 @@
+                value: "#foo:localhost",
+            }]));
+        });
+        it('accept #foo:bar.com', () => {
+            const test = '#foo:bar.com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
+            }]));
+        });
+        it('accept #foo:com (mostly for (TLD|DOMAIN)+ mixing)', () => {
+            const test = '#foo:com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:com",
+                type: "roomalias",
+                value: "#foo:com",
+            }]));
+        });
+        it('accept repeated TLDs (e.g .org.uk)', () => {
+            const test = '#foo:bar.org.uk';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.org.uk",
+                type: "roomalias",
+                value: "#foo:bar.org.uk",
+            }]));
+        });
+        it('ignores trailing `:`', () => {
+            const test = '#foo:bar.com:';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
+            }]));
+        });
+        it('accept :NUM (port specifier)', () => {
+            const test = '#foo:bar.com:2225';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com:2225",
+                type: "roomalias",
+                value: "#foo:bar.com:2225",
+            }]));
+        });
+        it('ignores all the trailing :', () => {
+            const test = '#foo:bar.com::::';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
+            }]));
+        });
+        it('properly parses room alias with dots in name', () => {
+            const test = '#foo.asdf:bar.com::::';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo.asdf:bar.com",
+                type: "roomalias",
+                value: "#foo.asdf:bar.com",
+
+            }]));
+        });
+        // This test does not work and does not parse the room
+        it.skip('does not parse room alias with too many separators', () => {
+            const test = '#foo:::bar.com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo",
+                type: 'roomalias',
+                value: '#foo',
+            },
+            {
+                href: "http://bar.com",
+                type: "url",
+                value: "bar.com",
+            }]));
+        });
```

Comment:
I would expect this to not parse the room though, `#foo` is not a valid roomalias.
This test is invalid.

Reply:
Fixed the test to make it valid instead